pandas/io/excel/_base.py

```diff
@@ -554,7 +554,9 @@ def parse(
                         header_name, _ = pop_header_name(data[row], index_col)
                         header_names.append(header_name)
 
-            has_index_names = is_list_like(header) and len(header) > 1
+            has_index_names = (
+                is_list_like(header) and len(header) > 1 and index_col is not None
```

Comment:
The docstring for TextParser says about has_index_names:

> True if the cols defined in index_col have an index name and are not in the header.

From this, it seems to me the inclusion of index_col you're doing here is correct. However I don't understand the other two conditions - can't we have index names when there is only one row for the header?

Reply:
From the [0.17 changes](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v0.17.0.html#changes-to-excel-with-multiindex), it appears that `has_index_names` was an argument for `read_excel` because the Excel output used to place the index name on its own row even for single-row headers. It was [dropped in 0.21](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v0.17.0.html#changes-to-excel-with-multiindex). The name is a bit confusing now. Maybe `has_index_name_row` would be better, but I'd have to check carefully before attempting a change. The are 14 lines on the code with `has_index_names` and some might have different meanings than this one.

Reply:
Thanks! To summarize, when header is a single row, the index names will not be on their own row. However, when header is multiple rows, the index names will be on their own row. Certainly wouldn't recommend changing the name here.